force-app/main/default/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -277,25 +294,32 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
                     numOfContactsLeft = accountIdNumOfCons.get(con.AccountId);
                 }
 
-                updateWrapper.handleUpdateData(con, oldContacts[conProcessed], triggerAction, numOfContactsLeft);
-            }
+                updateWrapper.handleUpdateData(
+                    con,
+                    oldContactByContactId.get(con.Id),
+                    triggerAction,
+                    numOfContactsLeft
+                );
 
-            conProcessed += 1;
-        }
+                conProcessed += 1;
+            }
 
-        //Set recursion flag
-        if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
             TDTM_ProcessControl.setRecursionFlag(
-                TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Insert,
+                TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Update,
                 true
             );
+            handleInsertProcessing(insertWrapper, contactsNeedAccounts);
```

Comment:
I was just curious to know: if the if statement already passed and we know for sure this is an afterUpdate, why would we need to call handleInsertProcessing()?

Reply:
We're making a minimum viable change here. So this is a single method refactor. In an ideal world we would evaluate the logic beyond this method and make a larger change. However, this is a small work item and the impact of goofing up this class can prevent contact inserts and updates, so we're playing this safe and small.

Reply:
sounds good, thanks for explaining the scenario.